### DISCUSSION TITLE
Issue with the labels off in taskbar and advanced mitigations for classic theme enabled

### DISCUSSION THREAD

**Discussion by OrthodoxWindows** [valinet/ExplorerPatcher#900] [General]
@valinet First of all, I want to thank you for your programs. It is absolutely great.

Secondly, i have a bug when the labels are hidden and the advanced mitigations for classic theme, and the classic theme, is enabled.
More precisely, the buttons seems too wide. 
It seems to me that the taskbar is set to occupy all available space :

![lTjjG9TbKe](https://user-images.githubusercontent.com/100134023/154955616-b48055d8-3496-4421-bfe1-4e74c09f718a.png)

With the labels displayed, the issue disappears :

![EOsvleZz5s](https://user-images.githubusercontent.com/100134023/154955573-0be61d50-80a6-4b5f-9b96-b62e625a968b.png)

The Explorer Patcher version is **22000.493.42.16**.

**Comment by valinet** [comment]
Yeah, it’s a known issue, when classic theme is enabled, the themes functions in the OS do not work; specifically stuff from uxtheme. Explorer uses some functions from there to calculate the size of the buttons. For classic theme with button labels, I fixed this manually (that’s what the mitigations are). For classic theme without button labels, this simply hasn’t been tackled by anyone, so it just is in its default, broken state.

**Comment by OrthodoxWindows** [comment]
I think this broken state is not only related to the default state, but also related to classic theme mitigations, because the the icons do not behave the same way with the classic theme without taskbar mitigations and OpenShell patched DLL.

Below is a comparison of the different states of the taskbar:

https://ncloud5.zaclys.com/index.php/s/ox9kt2PHkJowYiJ/preview
Taskbar with ExplorerPatcher with labels, with classic theme, without mitigations, without OpenShell : the labels are gone, and the color is not customizable.

https://ncloud5.zaclys.com/index.php/s/gT4agBxnomDa5H7/preview
Taskbar with ExplorerPatcher without labels, with classic theme, without mitigations, without OpenShell : the icons are too tight, and the color is not customizable.

https://ncloud5.zaclys.com/index.php/s/dS3F97y6LkPTyTw/preview
Taskbar with ExplorerPatcher with labels, with classic theme, without mitigations, with OpenShell, without OpenShell taskbar customization : the labels are gone, and the color is not customizable.

https://ncloud5.zaclys.com/index.php/s/g6qgxY6saAXYMiG/preview
Taskbar with ExplorerPatcher without labels, with classic theme, without mitigations, with OpenShell, without OpenShell taskbar customization : the icons are too tight, and the color is not customizable.

https://ncloud5.zaclys.com/index.php/s/KDTKk7f37DbiB5z/preview
Taskbar with ExplorerPatcher with labels, with classic theme, without mitigations, with OpenShell, with OpenShell taskbar customization : the labels are gone, the color can be selected to be the same as the windows, but this is not automatic.

https://ncloud5.zaclys.com/index.php/s/7XoLw4FroqQqGxJ/preview
Taskbar with ExplorerPatcher without labels, with classic theme, without mitigations, with OpenShell, with OpenShell taskbar customization : the icons the icons are too tight, and the color can be selected to be the same as the windows, but this is not automatic.

https://ncloud5.zaclys.com/index.php/s/CXGLMF9CjpYYS5n/preview
Taskbar with ExplorerPatcher without labels, with classic theme, with mitigations and OpenShell custom DLL, with OpenShell, with OpenShell taskbar customization : the labels are present, and the background color is aligned with the window color.

https://ncloud5.zaclys.com/index.php/s/iBkYFXE2LEdTobn/preview
Taskbar with ExplorerPatcher without labels, with classic theme,  with mitigations and OpenShell custom DLL, with OpenShell, without OpenShell taskbar customization : the background color is aligned with the window color, but the icons take up too much space.

**Reply by valinet** [reply]
Links do not work

**Reply by OrthodoxWindows** [reply]
I corrected, thank.

**Comment by valinet** [comment]
> Secondly, i have a bug when the labels are hidden and the advanced mitigations for classic theme, and the classic theme, is enabled.
More precisely, the buttons seems too wide. 

It's a bug in `explorer.exe` itself, or rather, it being unable to correctly query the sizes for the buttons because theme functions do not work. For labeled taskbar, that's what the "mitigations" in EP do: they fill in the gap and compute the widths another way so that `explorer.exe` gets what it wants. I couldn't make it work for label-less buttons though, hence the current broken state.
